pkg/controller.v1beta1/trial/trial_controller_test.go

```diff
@@ -134,59 +168,21 @@ func TestReconcileBatchJob(t *testing.T) {
 	viper.Set(consts.ConfigTrialResources, trialResources)
 	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
 
-	// Start test manager.
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
+	// Start test manager
+	mgrCtx, cancel := context.WithCancel(context.TODO())
```

Comment:
```suggestion
	mgrCtx, cancel := context.WithCancel(context.TODO())
	t.Cleanup(cancel)
```

Instead of calling cancel function in line 369, could you enforce to call cancel after this test is finished?

Reply:
Sure! I think it looks better to me.